tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -434,8 +446,21 @@ bool IsGpuCompatible(const RemapperContext& ctx,
                            filter_shape.dim(1).size() != 1;
 
     return is_spatial_conv && IsGpuCompatibleConv2D(ctx, &contraction_node);
-  } else if (IsMatMul(contraction_node)) {
-    return IsGpuCompatibleMatMul(ctx, &contraction_node);
+  } else if (IsMatMul(contraction_node) && supported_matmul_act) {
+    const std::vector<OpInfo::TensorProperties>& input_props =
+        ctx.graph_properties.GetInputProperties(contraction_node.name());
+    const TensorShapeProto& a_shape =
+        input_props.size() >= 1 ? input_props[0].shape() : TensorShapeProto();
+    const TensorShapeProto& b_shape =
+        input_props.size() >= 1 ? input_props[1].shape() : TensorShapeProto();
+    bool valid_dims = Rank(a_shape) == 2 && Rank(b_shape) == 2 &&
+                      IsKnown(a_shape.dim(1)) &&         //
+                      IsKnown(b_shape.dim(1)) &&         //
+                      a_shape.dim(1).size() % 2 == 0 &&  //
+                      b_shape.dim(1).size() % 2 == 0;
```

Comment:
Probably a good idea to add a comment explaining the `% 2 == 0` check like the one here:
https://github.com/tensorflow/tensorflow/blob/58213c8b32d9d5299484b63cdf153d4fb1d01a1a/tensorflow/core/grappler/optimizers/remapper.cc#L1278-L1279

Reply:
Done.